### PR TITLE
Add Drizzle tests

### PR DESCRIPTION
**Pull request by brunobar79**
This PR adds a drizzle app to our test suite as the initial step towards making sure we don't break web3 1.0 / drizzle support in the future.

For now the only test we have is just to check that we're exposing the address correctly but ideally we should check other basic stuff like send eth / tokens and ideally subs. Since all the workflow is already in place it should be pretty easy to add additional scenarios.

### CODE REVIEW

**Comment by whymarrh** [comment]
I think this has merit. I think that shouldn't have this in the same repository though as this will incur a large maintenance overhead.

I noticed that in #5450 Dan linked to his [issue-5425-repro](https://github.com/danfinlay/issue-5425-repro) repository—I think it might be best if we kept these as separate repositories and somehow introduced a generalized way of testing external dapps against each MM build. That way we could allow users to submit dapps as test cases. That way we could (one day) test against the top N dapps. Thoughts? 🤔

**Comment by brunobar79** [comment]
@whymarrh I understand your point and I think it’s fair.

 I should have added some emphasis that the main goal of this PR is not to test drizzle itself but to make sure we don’t break compatibility with web3 1.0 (which should be part of our test suite after the provider subs branch gets merged into dev).   We used drizzle because it’s probably the most popular framework using it, and yesterday while debugging the subs issue we agreed that it was necessary to add it to the test suite.

Regarding maintanability, there’s a package.json inside drizzle-test where we can bump major versions of truffle or drizzle. That should be all we need to maintain.


Pinging @danfinlay  @kumavis @thuang to hear their opinions.

**Comment by danfinlay** [comment]
I think you both have good points:
- This ensures we don't break Drizzle
- This adds a lot of overhead to the suite.

Part of me agrees with Whymarrh, thinking maybe we could just periodically run "the larger test suite" before a release, but that just postpones fixing new bugs. I tend to think this is kindof exactly what the test suite is for, and as long as we run it in parallel, it should be fine to add, right?

**Comment by whymarrh** [comment]
I think this is a fine addition, as my main (only?) concern was the bloat included from `truffle unbox drizzle` which is no longer an issue.

**Review by whymarrh — CHANGES_REQUESTED** [review]
Can we drop the changes to the `package-lock.json` file? They seem unneeded.

**Comment by brunobar79** [comment]
@whymarrh  done

File: test/e2e/beta/drizzle.spec.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+  const tinyDelayMs = 200
+  const regularDelayMs = tinyDelayMs * 2
+  const largeDelayMs = regularDelayMs * 2
+
+  this.timeout(0)
+  this.bail(true)
+
+  before(async function () {
+    switch (process.env.SELENIUM_BROWSER) {
```
Comment (kumavis) [inline test/e2e/beta/drizzle.spec.js]: This selenium bootstrapping should be extracted into a utility (I'm sure it already is somewhere). You shouldn't need more than a line or two for this generic setup

File: test/e2e/beta/run-all.sh [hunk 2]
```diff
@@ -6,5 +6,5 @@ set -o pipefail
 
 export PATH="$PATH:./node_modules/.bin"
 
-shell-parallel -s 'npm run ganache:start -- -b 2' -x 'sleep 5 && static-server test/e2e/beta/contract-test/ --port 8080' -x 'sleep 5 && mocha test/e2e/beta/metamask-beta-ui.spec'
-shell-parallel -s 'npm run ganache:start -- -d -b 2'  -x 'sleep 5 && static-server test/e2e/beta/contract-test/ --port 8080' -x 'sleep 5 && mocha test/e2e/beta/from-import-beta-ui.spec'
```
Comment (kumavis) [inline test/e2e/beta/run-all.sh]: Why this change?

File: test/e2e/beta/drizzle.spec.js [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+      await verboseReportOnFailure(driver, this.currentTest)
+    }
+  })
+
+  after(async function () {
+    await driver.quit()
+  })
+
+
+  describe('New UI setup', async function () {
```
Comment (kumavis) [inline test/e2e/beta/drizzle.spec.js]: All this generic init tests should also be abstracted into a library

File: test/e2e/beta/run-all.sh (hunk 2 above)
Comment (brunobar79) [reply test/e2e/beta/run-all.sh]: Because there’s nothing on that spec that requires the static server

File: test/e2e/beta/drizzle.spec.js (hunk 1 above)
Comment (brunobar79) [reply test/e2e/beta/drizzle.spec.js]: Sounds good! l’ll handle tomorrow.

**Review by whymarrh — APPROVED** [review]
I'm cool with these changes as-is. The suggestions @kumavis made could maybe be a separate PR since they'd touch other e2e tests.

**Comment by kumavis** [comment]
Agree with @whymarrh about making it a separate PR

**Comment by brunobar79** [comment]
@whymarrh  @kumavis Just created a new issue (#5488) to discuss with @tmashuang  and @danjm to find the best approach for this and will fix in a separate PR.
